src/test/codewhisperer/commands/basicCommands.test.ts

```diff
@@ -0,0 +1,121 @@
+/*!
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+import * as assert from 'assert'
+import { beforeEach } from 'mocha'
+import * as sinon from 'sinon'
+import { resetCodeWhispererGlobalVariables } from '../testUtil'
+import { assertTelemetryCurried } from '../../testUtil'
+import { CodeWhispererConstants } from '../../../codewhisperer/models/constants'
+import { toggleCodeSuggestions, get, set } from '../../../codewhisperer/commands/basicCommands'
+import { FakeExtensionContext, FakeMemento } from '../../fakeExtensionContext'
+import { ExtContext } from '../../../shared/extensions'
+import { AwsContext } from '../../../shared/awsContext'
+import { SamCliContext } from '../../../shared/sam/cli/samCliContext'
+import { RegionProvider } from '../../../shared/regions/regionProvider'
+import { TelemetryService } from '../../../shared/telemetry/telemetryService'
+import { CredentialsStore } from '../../../credentials/credentialsStore'
+import { testCommand } from '../../shared/vscode/testUtils'
+import { Command } from '../../../shared/vscode/commands2'
+
+describe('CodeWhisperer-basicCommands', function () {
+    describe('CodeWhisperer-basicCommands', function () {
+        let targetCommand: Command<any> & vscode.Disposable
+
+        beforeEach(function () {
+            resetCodeWhispererGlobalVariables()
+        })
+
+        afterEach(function () {
+            if (targetCommand) {
+                targetCommand.dispose()
+            }
+            sinon.restore()
+        })
+
+        it('test get()', async function () {
+            const fakeMemeto = new FakeMemento()
+            fakeMemeto.update(CodeWhispererConstants.autoTriggerEnabledKey, true)
+
+            let res = get(CodeWhispererConstants.autoTriggerEnabledKey, fakeMemeto)
+            assert.strictEqual(res, true)
+
+            fakeMemeto.update(CodeWhispererConstants.autoTriggerEnabledKey, undefined)
+            res = get(CodeWhispererConstants.autoTriggerEnabledKey, fakeMemeto)
+            assert.strictEqual(res, undefined)
+
+            fakeMemeto.update(CodeWhispererConstants.autoTriggerEnabledKey, false)
+            res = get(CodeWhispererConstants.autoTriggerEnabledKey, fakeMemeto)
+            assert.strictEqual(res, false)
+        })
+
+        it('test set()', async function () {
+            const fakeMemeto = new FakeMemento()
+            set(CodeWhispererConstants.autoTriggerEnabledKey, true, fakeMemeto)
+            assert.strictEqual(fakeMemeto.get(CodeWhispererConstants.autoTriggerEnabledKey), true)
+
+            set(CodeWhispererConstants.autoTriggerEnabledKey, false, fakeMemeto)
+            assert.strictEqual(fakeMemeto.get(CodeWhispererConstants.autoTriggerEnabledKey), false)
+        })
+
+        it('test toggleCodeSuggestions: should emit aws_modifySetting event on user toggling autoSuggestion -- activate', async function () {
```

Comment:
You had the right idea! This is a common pattern with the testing framework we use (`mocha`) and the convention is to make another describe block:
```suggestion
        describe('toggleCodeSuggestions', function () {
            it('should emit aws_modifySetting event on user toggling autoSuggestion -- activate', async function () {
```